sk0-front/src/app/auth/auth.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@angular/core';
+import { AlertController } from '@ionic/angular';
+
+type User = {
+  name: string;
+  email: string;
+  password: string;
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  private _isUserLoggedIn = false;
+  private _userName = 'Vendég';
+  private errorMessage: string | null = null;
+
+  // TODO: Connect FE with BE and remove mockUsers below
+  private mockUsers: User[];
+
+  get isUserLoggedIn() {
+    return this._isUserLoggedIn;
+  }
+
+  get userName() {
+    return this._userName;
+  }
+
+  constructor(private alertController: AlertController) {
+    // TODO: Connect FE with BE and remove mockUsers below
+
+    this.mockUsers = [
+      {
+        name: 'First User',
+        email: 'teszt@gmail.com',
+        password: 'Abcd1234!',
+      },
+      {
+        name: 'Second User',
+        email: 'seconduser@gmail.com',
+        password: 'Aaaa111!',
+      },
+    ];
+  }
+
+  async login(loginFormData: { email: string; password: string }) {
+    const { email, password } = loginFormData;
+    // TODO: Connect FE with BE and remove mockUsers and refactor code below
+    try {
+      const foundUser = this.mockUsers.find((user) => {
+        return user.email === email;
+      });
+      if (!foundUser) {
+        throw new Error('Nem megfelelő email cím vagy jelszó');
+      }
+      if (foundUser.password !== password) {
+        throw new Error('Nem megfelelő email cím vagy jelszó');
+      }
+    } catch (error) {
+      this.errorMessage = (error as Error).message;
+      await this.showError();
+      return;
```

Comment:
I think we should avoid "early void returns". If we want to return here, we can return with the error too. Or we might want to "collect" errors into an array and return at the end with it, but adding intermediate untested code it might leave to undefined behaviour.

Reply:
tl;dr 

```suggestion
   return this.showError();
```

Reply:
yess, i was not sure, thx!